### PR TITLE
add alternative to ClusterInfo::getCollection() that doesn't throw

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/2179/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/3293/ - merge overlap.

**Comment by dothebart** [comment]
Testrun success.

File: arangod/Utils/CollectionNameResolver.cpp [hunk 1]
```diff
@@ -118,18 +118,10 @@ TRI_voc_cid_t CollectionNameResolver::getCollectionIdCluster(
     // We have to look up the collection info:
     auto* ci = ClusterInfo::instance();
 
-    try {
-      auto const cinfo = ci->getCollection(_vocbase.name(), name);
+    auto const cinfo = (ci) ? ci->getCollectionNT(_vocbase.name(), name) : nullptr;
```
Comment (jsteemann) [inline arangod/Utils/CollectionNameResolver.cpp]: if `ci` can be a nullptr, then the program will crash when it reaches line 127
```
auto const vinfo = ci->getView(_vocbase.name(), name);
```

File: arangod/Utils/CollectionNameResolver.cpp [hunk 2]
```diff
@@ -374,17 +355,9 @@ std::shared_ptr<LogicalDataSource> CollectionNameResolver::getDataSource(
...
-        // FIXME by some reason 'ClusterInfo::getCollection' throws exception
-        // in case if collection is not found, ignore error
-        if (ex.code() != TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND) {
-          throw;
-        }
-      }
+      ptr = (ci) ? ci->getCollectionNT(_vocbase.name(), nameOrId) : nullptr;
 
-      if (!ptr) {
+      if (ptr == nullptr) {
```
Comment (jsteemann) [inline arangod/Utils/CollectionNameResolver.cpp]: if `ci` is null, then the program will crash in the following line

**Pull request merged**